### ISSUE TITLE
No output with RS485

### ISSUE THREAD

**Issue by d-tamm** [jbuehl/solaredge#19]
While trying to solve the network connection issues in #11, I have now connected to my SE5k inverter by RS485. Sadly, I am not getting any output. I have terminated the cable at the inverter side by moving up the switch in the inverter. At the computer side, the cable is currently unterminated because I do not really know what resistance my cable has (2x2 twisted wires phone cable, 1.5 m long).
Here is the debug.log I get by running `sudo python semonitor.py -m -s 7E1A0823 -d ../debug.log -vvvv -t 4 /dev/ttyUSB0`:
```
Aug 10 17:44:57 debugEnable: True 
Aug 10 17:44:57 debugFiles: True 
Aug 10 17:44:57 debugMsgs: True 
Aug 10 17:44:57 debugData: True 
Aug 10 17:44:57 debugRaw: True 
Aug 10 17:44:57 debugFileName: ../debug.log 
Aug 10 17:44:57 haltOnException: False 
Aug 10 17:44:57 inFileName: /dev/ttyUSB0 
Aug 10 17:44:57 inputType: 4 
Aug 10 17:44:57 serialDevice: True 
Aug 10 17:44:57     baudRate: 115200 
Aug 10 17:44:57 networkDevice: False 
Aug 10 17:44:57 networkSvcs: False 
Aug 10 17:44:57 following: True 
Aug 10 17:44:57 passiveMode: False 
Aug 10 17:44:57 commandAction: False 
Aug 10 17:44:57 masterMode: True 
Aug 10 17:44:57 slaveAddrs: 7E1A0823 
Aug 10 17:44:57 outFileName: stdout 
Aug 10 17:44:57 append: w 
Aug 10 17:44:57 opening /dev/ttyUSB0 
Aug 10 17:44:57 starting read thread 
Aug 10 17:44:57 starting master thread 
Aug 10 17:44:57 dataLen:    0000 
Aug 10 17:44:57 dataLenInv: ffff 
Aug 10 17:44:57 sequence:   0001 
Aug 10 17:44:57 source:     fffffffe 
Aug 10 17:44:57 dest:       7e1a0823 
Aug 10 17:44:57 function:   0302 
Aug 10 17:44:57 /dev/ttyUSB0 <-- message: 1 length: 22 
Aug 10 17:44:57 data:       12 34 56 79 00 00 ff ff 01 00 fe ff ff ff 23 08 
Aug 10 17:44:57 data:       1a 7e 02 03 fd 76 
Aug 10 17:44:57   
Aug 10 17:44:57   
Aug 10 17:44:57 /dev/ttyUSB0 --> message: 1 length: 22 
Aug 10 17:44:57 data:       12 34 56 79 00 00 ff ff 01 00 23 08 1a 7e fe ff 
Aug 10 17:44:57 data:       ff ff 9a 03 ee 43 
Aug 10 17:44:57 dataLen:    0000 
Aug 10 17:44:57 dataLenInv: ffff 
Aug 10 17:44:57 sequence:   0001 
Aug 10 17:44:57 source:     7e1a0823 
Aug 10 17:44:57 dest:       fffffffe 
Aug 10 17:44:57 function:   039a 
Aug 10 17:45:02 dataLen:    0000 
Aug 10 17:45:02 dataLenInv: ffff 
Aug 10 17:45:02 sequence:   0002 
Aug 10 17:45:02 source:     fffffffe 
Aug 10 17:45:02 dest:       7e1a0823 
Aug 10 17:45:02 function:   0302 
Aug 10 17:45:02 /dev/ttyUSB0 <-- message: 2 length: 22 
Aug 10 17:45:02 data:       12 34 56 79 00 00 ff ff 02 00 fe ff ff ff 23 08 
Aug 10 17:45:02 data:       1a 7e 02 03 f2 32 
Aug 10 17:45:02   
Aug 10 17:45:02   
Aug 10 17:45:02 /dev/ttyUSB0 --> message: 2 length: 22 
Aug 10 17:45:02 data:       12 34 56 79 00 00 ff ff 02 00 23 08 1a 7e fe ff 
Aug 10 17:45:02 data:       ff ff 9a 03 e1 07 
Aug 10 17:45:02 dataLen:    0000 
Aug 10 17:45:02 dataLenInv: ffff 
Aug 10 17:45:02 sequence:   0002 
Aug 10 17:45:02 source:     7e1a0823 
Aug 10 17:45:02 dest:       fffffffe 
Aug 10 17:45:02 function:   039a 
Aug 10 17:45:07 dataLen:    0000 
Aug 10 17:45:07 dataLenInv: ffff 
Aug 10 17:45:07 sequence:   0003 
Aug 10 17:45:07 source:     fffffffe 
Aug 10 17:45:07 dest:       7e1a0823 
Aug 10 17:45:07 function:   0302 
Aug 10 17:45:07 /dev/ttyUSB0 <-- message: 3 length: 22 
Aug 10 17:45:07 data:       12 34 56 79 00 00 ff ff 03 00 fe ff ff ff 23 08 
Aug 10 17:45:07 data:       1a 7e 02 03 f6 ce 
Aug 10 17:45:07   
Aug 10 17:45:07   
Aug 10 17:45:07 /dev/ttyUSB0 --> message: 3 length: 22 
Aug 10 17:45:07 data:       12 34 56 79 00 00 ff ff 03 00 23 08 1a 7e fe ff 
Aug 10 17:45:07 data:       ff ff 9a 03 e5 fb 
Aug 10 17:45:07 dataLen:    0000 
Aug 10 17:45:07 dataLenInv: ffff 
Aug 10 17:45:07 sequence:   0003 
Aug 10 17:45:07 source:     7e1a0823 
Aug 10 17:45:07 dest:       fffffffe 
Aug 10 17:45:07 function:   039a 
Aug 10 17:45:12 dataLen:    0000 
Aug 10 17:45:12 dataLenInv: ffff 
Aug 10 17:45:12 sequence:   0004 
Aug 10 17:45:12 source:     fffffffe 
Aug 10 17:45:12 dest:       7e1a0823 
Aug 10 17:45:12 function:   0302 
Aug 10 17:45:12 /dev/ttyUSB0 <-- message: 4 length: 22 
Aug 10 17:45:12 data:       12 34 56 79 00 00 ff ff 04 00 fe ff ff ff 23 08 
Aug 10 17:45:12 data:       1a 7e 02 03 ec ba 
Aug 10 17:45:12   
Aug 10 17:45:12   
Aug 10 17:45:12 /dev/ttyUSB0 --> message: 4 length: 22 
Aug 10 17:45:12 data:       12 34 56 79 00 00 ff ff 04 00 23 08 1a 7e fe ff 
Aug 10 17:45:12 data:       ff ff 9a 03 ff 8f 
Aug 10 17:45:12 dataLen:    0000 
Aug 10 17:45:12 dataLenInv: ffff 
Aug 10 17:45:12 sequence:   0004 
Aug 10 17:45:12 source:     7e1a0823 
Aug 10 17:45:12 dest:       fffffffe 
Aug 10 17:45:12 function:   039a 
Aug 10 17:45:17 dataLen:    0000 
Aug 10 17:45:17 dataLenInv: ffff 
Aug 10 17:45:17 sequence:   0005 
Aug 10 17:45:17 source:     fffffffe 
Aug 10 17:45:17 dest:       7e1a0823 
Aug 10 17:45:17 function:   0302 
Aug 10 17:45:17 /dev/ttyUSB0 <-- message: 5 length: 22 
Aug 10 17:45:17 data:       12 34 56 79 00 00 ff ff 05 00 fe ff ff ff 23 08 
Aug 10 17:45:17 data:       1a 7e 02 03 e8 46 
Aug 10 17:45:17   
Aug 10 17:45:17   
Aug 10 17:45:17 /dev/ttyUSB0 --> message: 5 length: 22 
Aug 10 17:45:17 data:       12 34 56 79 00 00 ff ff 05 00 23 08 1a 7e fe ff 
Aug 10 17:45:17 data:       ff ff 9a 03 fb 73 
Aug 10 17:45:17 dataLen:    0000 
Aug 10 17:45:17 dataLenInv: ffff 
Aug 10 17:45:17 sequence:   0005 
Aug 10 17:45:17 source:     7e1a0823 
Aug 10 17:45:17 dest:       fffffffe 
Aug 10 17:45:17 function:   039a 
Aug 10 17:45:22 dataLen:    0000 
Aug 10 17:45:22 dataLenInv: ffff 
Aug 10 17:45:22 sequence:   0006 
Aug 10 17:45:22 source:     fffffffe 
Aug 10 17:45:22 dest:       7e1a0823 
Aug 10 17:45:22 function:   0302 
Aug 10 17:45:22 /dev/ttyUSB0 <-- message: 6 length: 22 
Aug 10 17:45:22 data:       12 34 56 79 00 00 ff ff 06 00 fe ff ff ff 23 08 
Aug 10 17:45:22 data:       1a 7e 02 03 e7 02 
Aug 10 17:45:22   
Aug 10 17:45:22   
Aug 10 17:45:22 /dev/ttyUSB0 --> message: 6 length: 22 
Aug 10 17:45:22 data:       12 34 56 79 00 00 ff ff 06 00 23 08 1a 7e fe ff 
Aug 10 17:45:22 data:       ff ff 9a 03 f4 37 
Aug 10 17:45:22 dataLen:    0000 
Aug 10 17:45:22 dataLenInv: ffff 
Aug 10 17:45:22 sequence:   0006 
Aug 10 17:45:22 source:     7e1a0823 
Aug 10 17:45:22 dest:       fffffffe 
Aug 10 17:45:22 function:   039a 
Aug 10 17:45:27 dataLen:    0000 
Aug 10 17:45:27 dataLenInv: ffff 
Aug 10 17:45:27 sequence:   0007 
Aug 10 17:45:27 source:     fffffffe 
Aug 10 17:45:27 dest:       7e1a0823 
Aug 10 17:45:27 function:   0302 
Aug 10 17:45:27 /dev/ttyUSB0 <-- message: 7 length: 22 
Aug 10 17:45:27 data:       12 34 56 79 00 00 ff ff 07 00 fe ff ff ff 23 08 
Aug 10 17:45:27 data:       1a 7e 02 03 e3 fe 
Aug 10 17:45:27   
Aug 10 17:45:27   
Aug 10 17:45:27 /dev/ttyUSB0 --> message: 7 length: 22 
Aug 10 17:45:27 data:       12 34 56 79 00 00 ff ff 07 00 23 08 1a 7e fe ff 
Aug 10 17:45:27 data:       ff ff 9a 03 f0 cb 
Aug 10 17:45:27 dataLen:    0000 
Aug 10 17:45:27 dataLenInv: ffff 
Aug 10 17:45:27 sequence:   0007 
Aug 10 17:45:27 source:     7e1a0823 
Aug 10 17:45:27 dest:       fffffffe 
Aug 10 17:45:27 function:   039a 
Aug 10 17:45:32 dataLen:    0000 
Aug 10 17:45:32 dataLenInv: ffff 
Aug 10 17:45:32 sequence:   0008 
Aug 10 17:45:32 source:     fffffffe 
Aug 10 17:45:32 dest:       7e1a0823 
Aug 10 17:45:32 function:   0302 
Aug 10 17:45:32 /dev/ttyUSB0 <-- message: 8 length: 22 
Aug 10 17:45:32 data:       12 34 56 79 00 00 ff ff 08 00 fe ff ff ff 23 08 
Aug 10 17:45:32 data:       1a 7e 02 03 d3 ea 
Aug 10 17:45:32   
Aug 10 17:45:32   
Aug 10 17:45:32 /dev/ttyUSB0 --> message: 8 length: 22 
Aug 10 17:45:32 data:       12 34 56 79 00 00 ff ff 08 00 23 08 1a 7e fe ff 
Aug 10 17:45:32 data:       ff ff 9a 03 c0 df 
Aug 10 17:45:32 dataLen:    0000 
Aug 10 17:45:32 dataLenInv: ffff 
Aug 10 17:45:32 sequence:   0008 
Aug 10 17:45:32 source:     7e1a0823 
Aug 10 17:45:32 dest:       fffffffe 
Aug 10 17:45:32 function:   039a 
Aug 10 17:45:37 dataLen:    0000 
Aug 10 17:45:37 dataLenInv: ffff 
Aug 10 17:45:37 sequence:   0009 
Aug 10 17:45:37 source:     fffffffe 
Aug 10 17:45:37 dest:       7e1a0823 
Aug 10 17:45:37 function:   0302 
Aug 10 17:45:37 /dev/ttyUSB0 <-- message: 9 length: 22 
Aug 10 17:45:37 data:       12 34 56 79 00 00 ff ff 09 00 fe ff ff ff 23 08 
Aug 10 17:45:37 data:       1a 7e 02 03 d7 16 
Aug 10 17:45:37   
Aug 10 17:45:37   
Aug 10 17:45:37 /dev/ttyUSB0 --> message: 9 length: 22 
Aug 10 17:45:37 data:       12 34 56 79 00 00 ff ff 09 00 23 08 1a 7e fe ff 
Aug 10 17:45:37 data:       ff ff 9a 03 c4 23 
Aug 10 17:45:37 dataLen:    0000 
Aug 10 17:45:37 dataLenInv: ffff 
Aug 10 17:45:37 sequence:   0009 
Aug 10 17:45:37 source:     7e1a0823 
Aug 10 17:45:37 dest:       fffffffe 
Aug 10 17:45:37 function:   039a 
Aug 10 17:45:42 dataLen:    0000 
Aug 10 17:45:42 dataLenInv: ffff 
Aug 10 17:45:42 sequence:   000a 
Aug 10 17:45:42 source:     fffffffe 
Aug 10 17:45:42 dest:       7e1a0823 
Aug 10 17:45:42 function:   0302 
Aug 10 17:45:42 /dev/ttyUSB0 <-- message: 10 length: 22 
Aug 10 17:45:42 data:       12 34 56 79 00 00 ff ff 0a 00 fe ff ff ff 23 08 
Aug 10 17:45:42 data:       1a 7e 02 03 d8 52 
Aug 10 17:45:42   
Aug 10 17:45:42   
Aug 10 17:45:42 /dev/ttyUSB0 --> message: 10 length: 22 
Aug 10 17:45:42 data:       12 34 56 79 00 00 ff ff 0a 00 23 08 1a 7e fe ff 
Aug 10 17:45:42 data:       ff ff 9a 03 cb 67 
Aug 10 17:45:42 dataLen:    0000 
Aug 10 17:45:42 dataLenInv: ffff 
Aug 10 17:45:42 sequence:   000a 
Aug 10 17:45:42 source:     7e1a0823 
Aug 10 17:45:42 dest:       fffffffe 
Aug 10 17:45:42 function:   039a 
Aug 10 17:45:48 dataLen:    0000 
Aug 10 17:45:48 dataLenInv: ffff 
Aug 10 17:45:48 sequence:   000b 
Aug 10 17:45:48 source:     fffffffe 
Aug 10 17:45:48 dest:       7e1a0823 
Aug 10 17:45:48 function:   0302 
Aug 10 17:45:48 /dev/ttyUSB0 <-- message: 11 length: 22 
Aug 10 17:45:48 data:       12 34 56 79 00 00 ff ff 0b 00 fe ff ff ff 23 08 
Aug 10 17:45:48 data:       1a 7e 02 03 dc ae 
Aug 10 17:45:48   
Aug 10 17:45:48   
Aug 10 17:45:48 /dev/ttyUSB0 --> message: 11 length: 22 
Aug 10 17:45:48 data:       12 34 56 79 00 00 ff ff 0b 00 23 08 1a 7e fe ff 
Aug 10 17:45:48 data:       ff ff 9a 03 cf 9b 
Aug 10 17:45:48 dataLen:    0000 
Aug 10 17:45:48 dataLenInv: ffff 
Aug 10 17:45:48 sequence:   000b 
Aug 10 17:45:48 source:     7e1a0823 
Aug 10 17:45:48 dest:       fffffffe 
Aug 10 17:45:48 function:   039a 
Aug 10 17:45:53 dataLen:    0000 
Aug 10 17:45:53 dataLenInv: ffff 
Aug 10 17:45:53 sequence:   000c 
Aug 10 17:45:53 source:     fffffffe 
Aug 10 17:45:53 dest:       7e1a0823 
Aug 10 17:45:53 function:   0302 
Aug 10 17:45:53 /dev/ttyUSB0 <-- message: 12 length: 22 
Aug 10 17:45:53 data:       12 34 56 79 00 00 ff ff 0c 00 fe ff ff ff 23 08 
Aug 10 17:45:53 data:       1a 7e 02 03 c6 da 
Aug 10 17:45:53   
Aug 10 17:45:53   
Aug 10 17:45:53 /dev/ttyUSB0 --> message: 12 length: 22 
Aug 10 17:45:53 data:       12 34 56 79 00 00 ff ff 0c 00 23 08 1a 7e fe ff 
Aug 10 17:45:53 data:       ff ff 9a 03 d5 ef 
Aug 10 17:45:53 dataLen:    0000 
Aug 10 17:45:53 dataLenInv: ffff 
Aug 10 17:45:53 sequence:   000c 
Aug 10 17:45:53 source:     7e1a0823 
Aug 10 17:45:53 dest:       fffffffe 
Aug 10 17:45:53 function:   039a 
Aug 10 17:45:58 dataLen:    0000 
Aug 10 17:45:58 dataLenInv: ffff 
Aug 10 17:45:58 sequence:   000d 
Aug 10 17:45:58 source:     fffffffe 
Aug 10 17:45:58 dest:       7e1a0823 
Aug 10 17:45:58 function:   0302 
Aug 10 17:45:58 /dev/ttyUSB0 <-- message: 13 length: 22 
Aug 10 17:45:58 data:       12 34 56 79 00 00 ff ff 0d 00 fe ff ff ff 23 08 
Aug 10 17:45:58 data:       1a 7e 02 03 c2 26 
Aug 10 17:45:58   
Aug 10 17:45:58   
Aug 10 17:45:58 /dev/ttyUSB0 --> message: 13 length: 22 
Aug 10 17:45:58 data:       12 34 56 79 00 00 ff ff 0d 00 23 08 1a 7e fe ff 
Aug 10 17:45:58 data:       ff ff 9a 03 d1 13 
Aug 10 17:45:58 dataLen:    0000 
Aug 10 17:45:58 dataLenInv: ffff 
Aug 10 17:45:58 sequence:   000d 
Aug 10 17:45:58 source:     7e1a0823 
Aug 10 17:45:58 dest:       fffffffe 
Aug 10 17:45:58 function:   039a 
Aug 10 17:46:03 dataLen:    0000 
Aug 10 17:46:03 dataLenInv: ffff 
Aug 10 17:46:03 sequence:   000e 
Aug 10 17:46:03 source:     fffffffe 
Aug 10 17:46:03 dest:       7e1a0823 
Aug 10 17:46:03 function:   0302 
Aug 10 17:46:03 /dev/ttyUSB0 <-- message: 14 length: 22 
Aug 10 17:46:03 data:       12 34 56 79 00 00 ff ff 0e 00 fe ff ff ff 23 08 
Aug 10 17:46:03 data:       1a 7e 02 03 cd 62 
Aug 10 17:46:03   
Aug 10 17:46:03   
Aug 10 17:46:03 /dev/ttyUSB0 --> message: 14 length: 22 
Aug 10 17:46:03 data:       12 34 56 79 00 00 ff ff 0e 00 23 08 1a 7e fe ff 
Aug 10 17:46:03 data:       ff ff 9a 03 de 57 
Aug 10 17:46:03 dataLen:    0000 
Aug 10 17:46:03 dataLenInv: ffff 
Aug 10 17:46:03 sequence:   000e 
Aug 10 17:46:03 source:     7e1a0823 
Aug 10 17:46:03 dest:       fffffffe 
Aug 10 17:46:03 function:   039a 
Aug 10 17:46:08 dataLen:    0000 
Aug 10 17:46:08 dataLenInv: ffff 
Aug 10 17:46:08 sequence:   000f 
Aug 10 17:46:08 source:     fffffffe 
Aug 10 17:46:08 dest:       7e1a0823 
Aug 10 17:46:08 function:   0302 
Aug 10 17:46:08 /dev/ttyUSB0 <-- message: 15 length: 22 
Aug 10 17:46:08 data:       12 34 56 79 00 00 ff ff 0f 00 fe ff ff ff 23 08 
Aug 10 17:46:08 data:       1a 7e 02 03 c9 9e 
Aug 10 17:46:08   
Aug 10 17:46:08   
Aug 10 17:46:08 /dev/ttyUSB0 --> message: 15 length: 22 
Aug 10 17:46:08 data:       12 34 56 79 00 00 ff ff 0f 00 23 08 1a 7e fe ff 
Aug 10 17:46:08 data:       ff ff 9a 03 da ab 
Aug 10 17:46:08 dataLen:    0000 
Aug 10 17:46:08 dataLenInv: ffff 
Aug 10 17:46:08 sequence:   000f 
Aug 10 17:46:08 source:     7e1a0823 
Aug 10 17:46:08 dest:       fffffffe 
Aug 10 17:46:08 function:   039a 
Aug 10 17:46:13 dataLen:    0000 
Aug 10 17:46:13 dataLenInv: ffff 
Aug 10 17:46:13 sequence:   0010 
Aug 10 17:46:13 source:     fffffffe 
Aug 10 17:46:13 dest:       7e1a0823 
Aug 10 17:46:13 function:   0302 
Aug 10 17:46:13 /dev/ttyUSB0 <-- message: 16 length: 22 
Aug 10 17:46:13 data:       12 34 56 79 00 00 ff ff 10 00 fe ff ff ff 23 08 
Aug 10 17:46:13 data:       1a 7e 02 03 ad 4a 
Aug 10 17:46:13   
Aug 10 17:46:13   
Aug 10 17:46:13 /dev/ttyUSB0 --> message: 16 length: 22 
Aug 10 17:46:13 data:       12 34 56 79 00 00 ff ff 10 00 23 08 1a 7e fe ff 
Aug 10 17:46:13 data:       ff ff 9a 03 be 7f 
Aug 10 17:46:13 dataLen:    0000 
Aug 10 17:46:13 dataLenInv: ffff 
Aug 10 17:46:13 sequence:   0010 
Aug 10 17:46:13 source:     7e1a0823 
Aug 10 17:46:13 dest:       fffffffe 
Aug 10 17:46:13 function:   039a 
Aug 10 17:46:18 dataLen:    0000 
Aug 10 17:46:18 dataLenInv: ffff 
Aug 10 17:46:18 sequence:   0011 
Aug 10 17:46:18 source:     fffffffe 
Aug 10 17:46:18 dest:       7e1a0823 
Aug 10 17:46:18 function:   0302 
Aug 10 17:46:18 /dev/ttyUSB0 <-- message: 17 length: 22 
Aug 10 17:46:18 data:       12 34 56 79 00 00 ff ff 11 00 fe ff ff ff 23 08 
Aug 10 17:46:18 data:       1a 7e 02 03 a9 b6 
Aug 10 17:46:18   
Aug 10 17:46:18   
Aug 10 17:46:18 /dev/ttyUSB0 --> message: 17 length: 22 
Aug 10 17:46:18 data:       12 34 56 79 00 00 ff ff 11 00 23 08 1a 7e fe ff 
Aug 10 17:46:18 data:       ff ff 9a 03 ba 83 
Aug 10 17:46:18 dataLen:    0000 
Aug 10 17:46:18 dataLenInv: ffff 
Aug 10 17:46:18 sequence:   0011 
Aug 10 17:46:18 source:     7e1a0823 
Aug 10 17:46:18 dest:       fffffffe 
Aug 10 17:46:18 function:   039a 
Aug 10 17:46:23 dataLen:    0000 
Aug 10 17:46:23 dataLenInv: ffff 
Aug 10 17:46:23 sequence:   0012 
Aug 10 17:46:23 source:     fffffffe 
Aug 10 17:46:23 dest:       7e1a0823 
Aug 10 17:46:23 function:   0302 
Aug 10 17:46:23 /dev/ttyUSB0 <-- message: 18 length: 22 
Aug 10 17:46:23 data:       12 34 56 79 00 00 ff ff 12 00 fe ff ff ff 23 08 
Aug 10 17:46:23 data:       1a 7e 02 03 a6 f2 
Aug 10 17:46:23   
Aug 10 17:46:23   
Aug 10 17:46:23 /dev/ttyUSB0 --> message: 18 length: 22 
Aug 10 17:46:23 data:       12 34 56 79 00 00 ff ff 12 00 23 08 1a 7e fe ff 
Aug 10 17:46:23 data:       ff ff 9a 03 b5 c7 
Aug 10 17:46:23 dataLen:    0000 
Aug 10 17:46:23 dataLenInv: ffff 
Aug 10 17:46:23 sequence:   0012 
Aug 10 17:46:23 source:     7e1a0823 
Aug 10 17:46:23 dest:       fffffffe 
Aug 10 17:46:23 function:   039a 
Aug 10 17:46:28 dataLen:    0000 
Aug 10 17:46:28 dataLenInv: ffff 
Aug 10 17:46:28 sequence:   0013 
Aug 10 17:46:28 source:     fffffffe 
Aug 10 17:46:28 dest:       7e1a0823 
Aug 10 17:46:28 function:   0302 
Aug 10 17:46:28 /dev/ttyUSB0 <-- message: 19 length: 22 
Aug 10 17:46:28 data:       12 34 56 79 00 00 ff ff 13 00 fe ff ff ff 23 08 
Aug 10 17:46:28 data:       1a 7e 02 03 a2 0e 
Aug 10 17:46:28   
Aug 10 17:46:28   
Aug 10 17:46:28 /dev/ttyUSB0 --> message: 19 length: 22 
Aug 10 17:46:28 data:       12 34 56 79 00 00 ff ff 13 00 23 08 1a 7e fe ff 
Aug 10 17:46:28 data:       ff ff 9a 03 b1 3b 
Aug 10 17:46:28 dataLen:    0000 
Aug 10 17:46:28 dataLenInv: ffff 
Aug 10 17:46:28 sequence:   0013 
Aug 10 17:46:28 source:     7e1a0823 
Aug 10 17:46:28 dest:       fffffffe 
Aug 10 17:46:28 function:   039a 
Aug 10 17:46:33 dataLen:    0000 
Aug 10 17:46:33 dataLenInv: ffff 
Aug 10 17:46:33 sequence:   0014 
Aug 10 17:46:33 source:     fffffffe 
Aug 10 17:46:33 dest:       7e1a0823 
Aug 10 17:46:33 function:   0302 
Aug 10 17:46:33 /dev/ttyUSB0 <-- message: 20 length: 22 
Aug 10 17:46:33 data:       12 34 56 79 00 00 ff ff 14 00 fe ff ff ff 23 08 
Aug 10 17:46:33 data:       1a 7e 02 03 b8 7a 
Aug 10 17:46:33   
Aug 10 17:46:33   
Aug 10 17:46:33 /dev/ttyUSB0 --> message: 20 length: 22 
Aug 10 17:46:33 data:       12 34 56 79 00 00 ff ff 14 00 23 08 1a 7e fe ff 
Aug 10 17:46:33 data:       ff ff 9a 03 ab 4f 
Aug 10 17:46:33 dataLen:    0000 
Aug 10 17:46:33 dataLenInv: ffff 
Aug 10 17:46:33 sequence:   0014 
Aug 10 17:46:33 source:     7e1a0823 
Aug 10 17:46:33 dest:       fffffffe 
Aug 10 17:46:33 function:   039a 
Aug 10 17:46:38 dataLen:    0000 
Aug 10 17:46:38 dataLenInv: ffff 
Aug 10 17:46:38 sequence:   0015 
Aug 10 17:46:38 source:     fffffffe 
Aug 10 17:46:38 dest:       7e1a0823 
Aug 10 17:46:38 function:   0302 
Aug 10 17:46:38 /dev/ttyUSB0 <-- message: 21 length: 22 
Aug 10 17:46:38 data:       12 34 56 79 00 00 ff ff 15 00 fe ff ff ff 23 08 
Aug 10 17:46:38 data:       1a 7e 02 03 bc 86 
Aug 10 17:46:38   
Aug 10 17:46:38   
Aug 10 17:46:38 /dev/ttyUSB0 --> message: 21 length: 22 
Aug 10 17:46:38 data:       12 34 56 79 00 00 ff ff 15 00 23 08 1a 7e fe ff 
Aug 10 17:46:38 data:       ff ff 9a 03 af b3 
Aug 10 17:46:38 dataLen:    0000 
Aug 10 17:46:38 dataLenInv: ffff 
Aug 10 17:46:38 sequence:   0015 
Aug 10 17:46:38 source:     7e1a0823 
Aug 10 17:46:38 dest:       fffffffe 
Aug 10 17:46:38 function:   039a 
Aug 10 17:46:43 dataLen:    0000 
Aug 10 17:46:43 dataLenInv: ffff 
Aug 10 17:46:43 sequence:   0016 
Aug 10 17:46:43 source:     fffffffe 
Aug 10 17:46:43 dest:       7e1a0823 
Aug 10 17:46:43 function:   0302 
Aug 10 17:46:43 /dev/ttyUSB0 <-- message: 22 length: 22 
Aug 10 17:46:43 data:       12 34 56 79 00 00 ff ff 16 00 fe ff ff ff 23 08 
Aug 10 17:46:43 data:       1a 7e 02 03 b3 c2 
Aug 10 17:46:43   
Aug 10 17:46:43   
Aug 10 17:46:43 /dev/ttyUSB0 --> message: 22 length: 22 
Aug 10 17:46:43 data:       12 34 56 79 00 00 ff ff 16 00 23 08 1a 7e fe ff 
Aug 10 17:46:43 data:       ff ff 9a 03 a0 f7 
Aug 10 17:46:43 dataLen:    0000 
Aug 10 17:46:43 dataLenInv: ffff 
Aug 10 17:46:43 sequence:   0016 
Aug 10 17:46:43 source:     7e1a0823 
Aug 10 17:46:43 dest:       fffffffe 
Aug 10 17:46:43 function:   039a 
Aug 10 17:46:48 dataLen:    0000 
Aug 10 17:46:48 dataLenInv: ffff 
Aug 10 17:46:48 sequence:   0017 
Aug 10 17:46:48 source:     fffffffe 
Aug 10 17:46:48 dest:       7e1a0823 
Aug 10 17:46:48 function:   0302 
Aug 10 17:46:48 /dev/ttyUSB0 <-- message: 23 length: 22 
Aug 10 17:46:48 data:       12 34 56 79 00 00 ff ff 17 00 fe ff ff ff 23 08 
Aug 10 17:46:48 data:       1a 7e 02 03 b7 3e 
Aug 10 17:46:48   
Aug 10 17:46:48   
Aug 10 17:46:48 /dev/ttyUSB0 --> message: 23 length: 22 
Aug 10 17:46:48 data:       12 34 56 79 00 00 ff ff 17 00 23 08 1a 7e fe ff 
Aug 10 17:46:48 data:       ff ff 9a 03 a4 0b 
Aug 10 17:46:48 dataLen:    0000 
Aug 10 17:46:48 dataLenInv: ffff 
Aug 10 17:46:48 sequence:   0017 
Aug 10 17:46:48 source:     7e1a0823 
Aug 10 17:46:48 dest:       fffffffe 
Aug 10 17:46:48 function:   039a 
Aug 10 17:46:53 dataLen:    0000 
Aug 10 17:46:53 dataLenInv: ffff 
Aug 10 17:46:53 sequence:   0018 
Aug 10 17:46:53 source:     fffffffe 
Aug 10 17:46:53 dest:       7e1a0823 
Aug 10 17:46:53 function:   0302 
Aug 10 17:46:53 /dev/ttyUSB0 <-- message: 24 length: 22 
Aug 10 17:46:53 data:       12 34 56 79 00 00 ff ff 18 00 fe ff ff ff 23 08 
Aug 10 17:46:53 data:       1a 7e 02 03 87 2a 
Aug 10 17:46:53   
Aug 10 17:46:53   
Aug 10 17:46:53 /dev/ttyUSB0 --> message: 24 length: 22 
Aug 10 17:46:53 data:       12 34 56 79 00 00 ff ff 18 00 23 08 1a 7e fe ff 
Aug 10 17:46:53 data:       ff ff 9a 03 94 1f 
Aug 10 17:46:53 dataLen:    0000 
Aug 10 17:46:53 dataLenInv: ffff 
Aug 10 17:46:53 sequence:   0018 
Aug 10 17:46:53 source:     7e1a0823 
Aug 10 17:46:53 dest:       fffffffe 
Aug 10 17:46:53 function:   039a 
Aug 10 17:46:58 dataLen:    0000 
Aug 10 17:46:58 dataLenInv: ffff 
Aug 10 17:46:58 sequence:   0019 
Aug 10 17:46:58 source:     fffffffe 
Aug 10 17:46:58 dest:       7e1a0823 
Aug 10 17:46:58 function:   0302 
Aug 10 17:46:58 /dev/ttyUSB0 <-- message: 25 length: 22 
Aug 10 17:46:58 data:       12 34 56 79 00 00 ff ff 19 00 fe ff ff ff 23 08 
Aug 10 17:46:58 data:       1a 7e 02 03 83 d6 
Aug 10 17:46:58   
Aug 10 17:46:58   
Aug 10 17:46:58 /dev/ttyUSB0 --> message: 25 length: 22 
Aug 10 17:46:58 data:       12 34 56 79 00 00 ff ff 19 00 23 08 1a 7e fe ff 
Aug 10 17:46:58 data:       ff ff 9a 03 90 e3 
Aug 10 17:46:58 dataLen:    0000 
Aug 10 17:46:58 dataLenInv: ffff 
Aug 10 17:46:58 sequence:   0019 
Aug 10 17:46:58 source:     7e1a0823 
Aug 10 17:46:58 dest:       fffffffe 
Aug 10 17:46:58 function:   039a 
Aug 10 17:47:03 dataLen:    0000 
Aug 10 17:47:03 dataLenInv: ffff 
Aug 10 17:47:03 sequence:   001a 
Aug 10 17:47:03 source:     fffffffe 
Aug 10 17:47:03 dest:       7e1a0823 
Aug 10 17:47:03 function:   0302 
Aug 10 17:47:03 /dev/ttyUSB0 <-- message: 26 length: 22 
Aug 10 17:47:03 data:       12 34 56 79 00 00 ff ff 1a 00 fe ff ff ff 23 08 
Aug 10 17:47:03 data:       1a 7e 02 03 8c 92 
Aug 10 17:47:03   
Aug 10 17:47:03   
Aug 10 17:47:03 /dev/ttyUSB0 --> message: 26 length: 22 
Aug 10 17:47:03 data:       12 34 56 79 00 00 ff ff 1a 00 23 08 1a 7e fe ff 
Aug 10 17:47:03 data:       ff ff 9a 03 9f a7 
Aug 10 17:47:03 dataLen:    0000 
Aug 10 17:47:03 dataLenInv: ffff 
Aug 10 17:47:03 sequence:   001a 
Aug 10 17:47:03 source:     7e1a0823 
Aug 10 17:47:03 dest:       fffffffe 
Aug 10 17:47:03 function:   039a 
Aug 10 17:47:08 dataLen:    0000 
Aug 10 17:47:08 dataLenInv: ffff 
Aug 10 17:47:08 sequence:   001b 
Aug 10 17:47:08 source:     fffffffe 
Aug 10 17:47:08 dest:       7e1a0823 
Aug 10 17:47:08 function:   0302 
Aug 10 17:47:08 /dev/ttyUSB0 <-- message: 27 length: 22 
Aug 10 17:47:08 data:       12 34 56 79 00 00 ff ff 1b 00 fe ff ff ff 23 08 
Aug 10 17:47:08 data:       1a 7e 02 03 88 6e 
Aug 10 17:47:08   
Aug 10 17:47:08   
Aug 10 17:47:08 /dev/ttyUSB0 --> message: 27 length: 22 
Aug 10 17:47:08 data:       12 34 56 79 00 00 ff ff 1b 00 23 08 1a 7e fe ff 
Aug 10 17:47:08 data:       ff ff 9a 03 9b 5b 
Aug 10 17:47:08 dataLen:    0000 
Aug 10 17:47:08 dataLenInv: ffff 
Aug 10 17:47:08 sequence:   001b 
Aug 10 17:47:08 source:     7e1a0823 
Aug 10 17:47:08 dest:       fffffffe 
Aug 10 17:47:08 function:   039a 
Aug 10 17:47:13 dataLen:    0000 
Aug 10 17:47:13 dataLenInv: ffff 
Aug 10 17:47:13 sequence:   001c 
Aug 10 17:47:13 source:     fffffffe 
Aug 10 17:47:13 dest:       7e1a0823 
Aug 10 17:47:13 function:   0302 
Aug 10 17:47:13 /dev/ttyUSB0 <-- message: 28 length: 22 
Aug 10 17:47:13 data:       12 34 56 79 00 00 ff ff 1c 00 fe ff ff ff 23 08 
Aug 10 17:47:13 data:       1a 7e 02 03 92 1a 
Aug 10 17:47:13   
```
No other output with production data whatsoever. Anybody knows what functions 039a or 0302 mean? Where may this information be found?
Also, the transmission stalls after some time, however not reproducibly. Could this be due to the missing termination and the stability issue named in #18 ?

**Comment by ocaj-nl** [comment]
What you see is the basic RS485-solaredge protocol where the master allows the inverter to communicate. If the inverter has nothing to say, it will return bus-access to the master.

I had this too initially and the most obvious (once you know it...) reason is that you did not configure the inverter to use RS485 as Server, but your inverter is still set to <LAN> as Server. If that is the case, the inverter will not send production infromation over RS485.

You can configure the server in the inverter-menu.

**Comment by d-tamm** [comment]
Oh, this is really great! Indeed, the inverter was set to LAN as server (I did not even know that it had such a setting). Now I receive production data. Thanks a lot for the fast reply @ocaj-nl !
By the way, does this mean that I cannot use the RS485 in parallel with SolarEdge's monitoring portal?

**Comment by ocaj-nl** [comment]
I personally have no interest in connecting my inverters to Solaredge's monitoring portal, so I intend to do RS485-only.  

I do not know whether the inverter supports reporting to 2 channels, I don't think it does. Check the solaredge-documentation, it has multiple connection scenarios described.

**Comment by t-beech** [comment]
As ocaj-nl already pointed out, this should be described [here](http://www.solaredge.com/sites/default/files/se-communication-options-application-note.pdf). 
I would go for Multiple Inverters, RS485 Bus, Wired Ethernet, Parallel Monitoring which states
that the inverters can be set to enable MODBUS in parallel to LAN for communication with the
SolarEdge servers. I have not been able to test, since I am waiting for my pv system to be installed.

**Comment by d-tamm** [comment]
Thanks for pointing out the documentation @t-beech . It is not really important for me to send to SolarEdge in parallel, I was just curious. I wonder also if the configuration you propose just adhers to the SunSpec protocol, which would imply that no data on optimizer level would be available on RS485. If I get some time, I might test this out, but it is low priority for me.

**Comment by JacquesWattel** [comment]
I am having similar issues. I am able to query individual parameters using RS485, but with the above command: sudo python semonitor.py -m -s 7E1A0823 -d ../debug.log -vvvv -t 4 /dev/ttyUSB0 I receive a RS485 master ack timeout error. Any suggestions?

**Comment by JacquesWattel** [comment]
Sorted out the problem, I as receiving data now via RS485 to a Raspberry PI. The problem now is that the data is very intermitted. Is the problem perhaps with the Raspberry PI?

**Comment by d-tamm** [comment]
Would you mind sharing what you changed in order to get rid of the timeouts?

**Comment by JacquesWattel** [comment]
Swapped the RS485 data line around. Silly mistake.

**Comment by jbuehl** [comment]
What do you mean that the data is intermittent, @JacquesWattel?  If you look at the debug data you should see a message from the Raspberry Pi every 5 seconds followed by a message from the inverter, but this does not always contain performance data.  This is only sent every minute or so, depending on how much power is being produced.  This is normal behavior.
